sources/ghostdog/src/main.rs

```diff
@@ -63,6 +141,44 @@ fn run() -> Result<()> {
             let device_name = find_device_name(format!("{}", path.display()))?;
             emit_device_name(&device_name);
         }
+        SubCommand::MatchNvidiaDriver(vendor) => {
+            // Currently, we only want to run this for NVIDIA devices
+            if vendor.vendor == NVIDIA_VENDOR_ID {
+                let pci_id = env::var("PCI_ID").context(error::MissingPciIdEnvSnafu)?;
+                let driver_choice = find_preferred_driver(pci_id)?;
+                // After choosing the driver, drop a marker file provide this decision to others on the system (like systemd)
+                if !Path::new(NVIDIA_RUN_DIR).exists() {
+                    fs::create_dir_all(NVIDIA_RUN_DIR).context(error::CreateDirSnafu {
+                        path: NVIDIA_RUN_DIR,
+                    })?;
+                }
+                let marker_path: PathBuf = Path::new(NVIDIA_RUN_DIR).join(driver_choice.clone());
+                fs::write(marker_path.clone(), "").context(error::WriteMarkerFileSnafu {
+                    path: marker_path.clone(),
+                })?;
```

Comment:
I didn't understand at first why this marker is necessary - `nvidia-open-gpu-modules.service` should only be active if this udev rule adds it.

I've left more comments on that service, but I think it'd be better to avoid doing this and shift more work into `driverdog`.

Reply:
Moved away from this approach, `ghostdog` only worries about checking the driver and exits with a code to represent that.